src/python/pants/engine/legacy/graph.py

```diff
@@ -296,6 +389,58 @@ class HydratedTargets(Collection.of(HydratedTarget)):
   """An intransitive set of HydratedTarget objects."""
 
 
+class OwnersRequest(datatype([
+  ('sources', tuple),
+  ('include_dependees', str),
+])):
+  """A request for the owners (and optionally, transitive dependees) of a set of file paths."""
+
+
+def find_owners(symbol_table, address_mapper, owners_request):
+  sources_set = set(owners_request.sources)
+  dirs_set = {dirname(source) for source in sources_set}
+  # Walk up the buildroot looking for targets that would conceivably claim changed sources.
+  candidate_specs = tuple(AscendantAddresses(directory=d) for d in dirs_set)
```

Comment:
This is not used by `owns_any_source`. How about moving it together with it's one use to yield hydrated_targets. IE: Move `owns_any_source` down in-between `hydrated_targets = ...` and `direct_owners = ...`.